### PR TITLE
feat: port RingTheory.Polynomial.Quotient

### PR DESCRIPTION
**Pull request by Ruben-VandeVelde**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by jjaassoonn** [comment]
```lean
noncomputable def quotientSpanXSubCAlgEquivAux2 (x : R) :
    (R[X] ⧸ (RingHom.ker (aeval x).toRingHom : Ideal R[X])) ≃ₐ[R] R :=
let e := RingHom.quotientKerEquivOfRightInverse (by
    intro x; exact eval_C : Function.RightInverse (fun a : R => (C a : R[X])) (@aeval R R _ _ _ x))
{ e with commutes' := fun r => e.apply_symm_apply r }

noncomputable def quotientSpanXSubCAlgEquivAux1 (x : R) :
  (R[X] ⧸ Ideal.span {X - C x}) ≃ₐ[R] (R[X] ⧸ (RingHom.ker (aeval x).toRingHom : Ideal R[X])) :=
@Ideal.quotientEquivAlgOfEq R R[X] _ _ _ _ _ (ker_evalRingHom x).symm

noncomputable def quotientSpanXSubCAlgEquiv (x : R) :
    (R[X] ⧸ Ideal.span ({X - C x} : Set R[X])) ≃ₐ[R] R :=
  (quotientSpanXSubCAlgEquivAux1 x).trans (quotientSpanXSubCAlgEquivAux2 x)
```

This works but introduces two sub definitions, so I am not sure if I should  push it?

**Comment by Ruben-VandeVelde** [comment]
Feel free to push

File: Mathlib/RingTheory/Polynomial/Quotient.lean [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+let e :=
+  RingHom.quotientKerEquivOfRightInverse (by intro x; exact eval_C :
+    Function.RightInverse (fun a : R => (C a : R[X])) (@aeval R R _ _ _ x))
+{ e with commutes' := fun r => e.apply_symm_apply r }
+
+noncomputable def quotientSpanXSubCAlgEquivAux1 (x : R) :
+  (R[X] ⧸ Ideal.span {X - C x}) ≃ₐ[R] (R[X] ⧸ (RingHom.ker (aeval x).toRingHom : Ideal R[X])) :=
+@Ideal.quotientEquivAlgOfEq R R[X] _ _ _ _ _ (ker_evalRingHom x).symm
+
+noncomputable def quotientSpanXSubCAlgEquiv (x : R) :
```
Comment (jjaassoonn) [inline Mathlib/RingTheory/Polynomial/Quotient.lean:40]: As mentioned above, this has been splited into two sub-constructions which may or may not be ideal

**Comment by semorrison** [comment]
Closing in favour of #3932.

**Pull request closed without merging**